### ISSUE TITLE
Config file, please

### ISSUE THREAD

**Issue by ThorlarTwitch** [Doug-Murphy/LethalCompanyMods#2]
Hi! I hope this is the right place to make this request, couldn't find you on the modder Discord for Lethal Company.

I love the concept of this mod and many changes it does, however there is a huge interest in my Twitch community (700+ people) to see this get a config file. I'd personally love to adjust a few things, and I can also see this explode like crazy as a mod if or when a config file becomes available!

Thank you for this lovely mod and looking forward to see it evolve! <3

**Comment by Doug-Murphy** [comment]
It definitely is the right place to make this request! And you're not the first to request a config. It is definitely on the roadmap. I'm currently in the late stages of 1.1.0 with adding value overrides. I made a lot of progress working on that last night and will very likely release it this weekend.

Config file will follow immediately after that, or maybe even alongside 1.1.0.

Thanks for the feedback and encouragement!

**Comment by Doug-Murphy** [comment]
@ThorlarTwitch I have a beta release available for config file support. Before I release the update to Thunderstore, I'm curious if you would like to playtest this for me and provide feedback? If so, I would appreciate it! If not, that's ok too. :)

**Comment by Doug-Murphy** [comment]
Awesome! Thanks a lot for volunteering. Below are instructions for downloading and installing it.

# Downloading the beta version
1. Navigate to https://github.com/Doug-Murphy/LethalCompanyMods/releases/tag/1.2.0-beta
2. Download the file named GuysNight.LethalCompanyMod.BalancedItems.dll

# Installing the beta version
1. Open r2modman
3. Select the profile you have for Lethal Company.
4. Disable the existing Balanced_Items mod. (We want to make sure there are no funky conflicts. I'm not sure how r2 or BepInEx would handle this)
5. Navigate to "Settings" under "OTHER" in the left navigation menu.
6. Find the "Browse profile folder" option and click that.
7. Open the `BepInEx` folder, then the `plugins` folder.
8. Drop the updated DLL that you downloaded from above into this folder.
9. Back in r2modman, start Lethal Company modded as you normally do.

# Config file
The config file will generate dynamically as you play the game (see more info below). So do not expect it to be there as soon as you launch Lethal Company. The config file can be edited in r2modman in the same way as all other config files, or you can manually edit it by finding it in the `BepInEx\config` folder (instead of `BepInEx\plugins` folder above). It will be named `GuysNight.LethalCompanyMod.BalancedItems.cfg`

I tuned into your stream for a bit this morning. Entertaining content! :) I saw that you have some mods that add in some custom items. I was glad to see that because the way I implemented the config file it should support letting you override those weights and values as well! I would really appreciate if you could confirm the config file generates entries for those items and lets you override their weight and average sell values!

A few things to keep in mind:
1. This is a beta release, which means the config file stuff could change. So after you finish your playtesting and before updating to the official release when it comes, it may be worthwhile to delete the generated .cfg file just in case.
2. The config file generation is fairly unique based on what I've seen from the modding community. I recommend reading [the documentation](https://github.com/Doug-Murphy/LethalCompanyMods/blob/1.2.0-beta/GuysNight.LethalCompanyMod.BalancedItems/README.md#configuration) to see how it works.
3. When you are done playtesting, just delete the DLL file that you put in the `BepInEx\plugins` folder to remove this beta version. The only other lingering artifact would be the config file.
4. Any items that have already spawned, or are currently on your ship, will not have their sell values adjusted. Weight values should adjust though. :) (This isn't really new in 1.2.0, but I am noting it just to be clear).

Thanks for playtesting this for me! I'm looking forward to hearing the feedback. :)

**Comment by Doug-Murphy** [comment]
I just pushed a few minor QoL improvements. You'll find the latest at https://github.com/Doug-Murphy/LethalCompanyMods/releases/tag/1.2.0-beta2

**Comment by TheRealMrKam** [comment]
Hey ! i tested it a bit, it's super cool because it seem to detect modded scrap items well.
But it seems buyable items (like shovel, flashlight...) aren't on the config, only the scrap items !
(Tested on v40 with other mods)

**Comment by Doug-Murphy** [comment]
They should pop into the config when they spawn in. So if you buy some, or have them in your ship when you load in, they should get added to the config. Did you try that out?

**Comment by TheRealMrKam** [comment]
Oh! My bad, it did appear now. I put the weight to 50 but when i grab it the weight is 52 for some reason.

**Comment by Doug-Murphy** [comment]
> I put the weight to 50 but when i grab it the weight is 52 for some reason.

Yeah there's some info about that in the readme. Super odd behavior from the game. I'd like to chalk it up to issues with rounding float values, but that should only give a +/- 1 weight per item, but as you see sometimes it's +2... I have no clue why the game does that. 🤷🏻‍♂️

**Comment by Doug-Murphy** [comment]
@qwbif that is an idea I had actually. When I was changing values, I saw the rarity in there too. That is something that's going to take some research though to understand. If you check the logs, you'll see the rarity being things like 5 and 18 and 3, but what does that actually mean? Maybe it's relative weights within that level? Not sure yet! Haha. That would be a good 1.3.0 feature. Can you open a new issue to track that specifically please? Thanks!

**Comment by Doug-Murphy** [comment]
@TheRealMrKam @ThorlarTwitch I'd like to release 1.2.0 on Thursday or Friday. Will that be enough time to finish your play testing and gathering your thoughts?

**Comment by Doug-Murphy** [comment]
@TheRealMrKam @ThorlarTwitch I just released a beta3 version of 1.2.0.
https://github.com/Doug-Murphy/LethalCompanyMods/releases/tag/1.2.0-beta3

@TheRealMrKam the notable change here is you no longer have to buy equipment and have it spawn into the world for it to be added to the config. :) Thanks for the inspiration to make me make that process not suck!

**Comment by Doug-Murphy** [comment]
Apologies for the spam but the ideas keep flowing. Yet another beta release here. Release notes and DLL are in the link below.

https://github.com/Doug-Murphy/LethalCompanyMods/releases/tag/1.2.0-beta4

**Comment by Doug-Murphy** [comment]
@ThorlarTwitch I watched a VOD of your stream from this morning and saw that you were having some issues finding items in your config using r2. I addressed this by sorting the items alphabetically before it adds items to the config file!

Furthermore, when you made the shovel 500 pounds and it displayed 525, that was kind of the last straw for me in just ignoring that bug in the game. So I did some digging tonight and found out where that carry weight calculation is done to display it on the HUD and I fixed it. Huzzah!

Latest DLL can be downloaded from the link below!
https://github.com/Doug-Murphy/LethalCompanyMods/releases/tag/1.2.0-beta5
